File: web-bff/src/routes/games.js
const express = require("express");

const {
    getGames,
    getGame
} = require("../gameCatalogClient");

const router = express.Router();

router.get("/games", async (req, res) => {

    try {

        const games = await getGames();

        res.json(games);

    } catch (error) {

        console.error(
            "Game Catalog error:",
            error.message
        );

        res.status(502).json({
            error: "Game Catalog Service unavailable"
        });
    }
});

router.get("/games/:id", async (req, res) => {

    try {

        const game = await getGame(req.params.id);

        res.json(game);

    } catch (error) {

        res.status(502).json({
            error: "Game Catalog Service unavailable"
        });
    }
});

module.exports = router;

File: mobile-bff/app/games.py
import os
import requests


GAME_CATALOG_URL = os.getenv(
    "GAME_CATALOG_URL",
    "http://localhost:8081"
)


def get_games():

    print("Mobile BFF -> Game Catalog")

    response = requests.get(
        f"{GAME_CATALOG_URL}/games"
    )

    response.raise_for_status()

    return response.json()


def get_game(game_id):

    print(
        f"Mobile BFF -> Game Catalog: {game_id}"
    )

    response = requests.get(
        f"{GAME_CATALOG_URL}/games/{game_id}"
    )

    response.raise_for_status()

    return response.json()